apps/dashboard/src/main/java/com/akto/utils/RiskScoreOfCollections.java

```diff
@@ -0,0 +1,182 @@
+package com.akto.utils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+
+import org.bson.conversions.Bson;
+
+import com.akto.action.observe.InventoryAction;
+import com.akto.dao.AccountSettingsDao;
+import com.akto.dao.AktoDataTypeDao;
+import com.akto.dao.ApiInfoDao;
+import com.akto.dao.CustomDataTypeDao;
+import com.akto.dao.SingleTypeInfoDao;
+import com.akto.dao.context.Context;
+import com.akto.dao.testing_run_findings.TestingRunIssuesDao;
+import com.akto.dto.AccountSettings;
+import com.akto.dto.ApiInfo;
+import com.akto.dto.AccountSettings.CronTimers;
+import com.akto.dto.AktoDataType;
+import com.akto.dto.ApiInfo.ApiInfoKey;
+import com.akto.dto.CustomDataType;
+import com.akto.dto.test_run_findings.TestingIssuesId;
+import com.akto.dto.test_run_findings.TestingRunIssues;
+import com.akto.dto.type.SingleTypeInfo;
+import com.mongodb.BasicDBObject;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.model.Aggregates;
+import com.mongodb.client.model.BulkWriteOptions;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Projections;
+import com.mongodb.client.model.UpdateManyModel;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.model.Updates;
+import com.mongodb.client.model.WriteModel;
+
+public class RiskScoreOfCollections {
+
+    public void updateSeverityScoreInApiInfo(int timeStampFilter){
+        Bson projections = Projections.include("_id", "lastSeen");
+        ArrayList<WriteModel<ApiInfo>> bulkUpdatesForApiInfo = new ArrayList<>();
+        
+        Bson filters = Filters.gte("lastSeen", timeStampFilter);
+        List<TestingRunIssues> issues = TestingRunIssuesDao.instance.findAll(filters, projections);
+        if(issues.size() > 0){
+            List<ApiInfoKey> updatedApiInfoKeys = new ArrayList<>();
+
+            for(TestingRunIssues issue: issues){
+                TestingIssuesId issueId = issue.getId();
+                ApiInfoKey apiInfoKey = issueId.getApiInfoKey();
+                updatedApiInfoKeys.add(apiInfoKey);
+            }
+
+            Bson filterQ = Filters.and(
+                Filters.eq(TestingRunIssues.TEST_RUN_ISSUES_STATUS, "OPEN"),
+                Filters.in("_id.apiInfoKey", updatedApiInfoKeys)
+            );
+            try {
```

Comment:
not a good practice to keep such global try catch blocks. identify where this function might break, and place accordingly. Would be easier to identify bugs later on

Reply:
done